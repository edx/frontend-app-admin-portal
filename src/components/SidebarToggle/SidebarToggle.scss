.sidebar-toggle-btn {
    width: 40px;
}
