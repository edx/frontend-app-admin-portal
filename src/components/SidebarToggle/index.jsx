import React from 'react';
import PropTypes from 'prop-types';
import { Button, Icon } from '@edx/paragon';
import classNames from 'classnames';


function SidebarToggle(props) {
  return (
    <Button
      label={
        <span>
          <Icon
            className={
              classNames({
                fa: true,
                'fa-times': props.expanded,
                'fa-bars': !props.expanded,
              }).split(' ')
            }
          />
        </span>
      }
      onClick={props.toggleSidebar}
    />
  );
}

SidebarToggle.defaultProps = {
  expanded: false,
  toggleSidebar: () => {},
};

SidebarToggle.propTypes = {
  expanded: PropTypes.bool,
  toggleSidebar: PropTypes.func,
};

export default SidebarToggle;
