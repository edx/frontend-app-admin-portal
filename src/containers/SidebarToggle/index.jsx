import { connect } from 'react-redux';

import SidebarToggle from '../../components/SidebarToggle';

import {
  expandSidebar,
  collapseSidebar,
} from '../../data/actions/sidebar';

const mapStateToProps = state => ({
  isExpandedByToggle: state.sidebar.isExpandedByToggle,
});

const mapDispatchToProps = dispatch => ({
  expandSidebar: () => dispatch(expandSidebar(true)),
  collapseSidebar: () => dispatch(collapseSidebar(true)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SidebarToggle);
