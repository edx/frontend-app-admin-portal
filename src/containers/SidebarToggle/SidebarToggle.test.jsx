import React from 'react';
import PropTypes from 'prop-types';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import SidebarToggle from './index';

import {
  EXPAND_SIDEBAR,
  COLLAPSE_SIDEBAR,
} from '../../data/constants/sidebar';

const mockStore = configureMockStore([thunk]);
const initialState = {
  sidebar: {
    isExpandedByToggle: false,
  },
};

const SidebarToggleWrapper = props => (
  <Provider store={props.store}>
    <SidebarToggle
      baseUrl="/test-enterprise-slug"
      {...props}
    />
  </Provider>
);

SidebarToggleWrapper.defaultProps = {
  store: mockStore({
    ...initialState,
  }),
};

SidebarToggleWrapper.propTypes = {
  store: PropTypes.shape({}),
};

describe('<Sidebar />', () => {
  it('renders correctly with menu icon', () => {
    const tree = renderer
      .create((
        <SidebarToggleWrapper />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with close icon', () => {
    const store = mockStore({
      sidebar: {
        ...initialState.sidebar,
        isExpandedByToggle: true,
      },
    });

    const tree = renderer
      .create((
        <SidebarToggleWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatches expandSidebar action', () => {
    const store = mockStore({
      ...initialState,
    });

    const wrapper = mount((
      <SidebarToggleWrapper store={store} />
    ));

    const expectedActions = [{
      type: EXPAND_SIDEBAR,
      payload: { usingToggle: true },
    }];

    store.clearActions();
    wrapper.find('SidebarToggle').simulate('click');
    expect(store.getActions()).toEqual(expectedActions);
  });

  it('dispatches collapseSidebar action', () => {
    const store = mockStore({
      ...initialState,
      sidebar: {
        ...initialState.sidebar,
        isExpandedByToggle: true,
      },
    });

    const wrapper = mount((
      <SidebarToggleWrapper store={store} />
    ));

    const expectedActions = [{
      type: COLLAPSE_SIDEBAR,
      payload: { usingToggle: true },
    }];

    store.clearActions();
    wrapper.find('SidebarToggle').simulate('click');
    expect(store.getActions()).toEqual(expectedActions);
  });
});
