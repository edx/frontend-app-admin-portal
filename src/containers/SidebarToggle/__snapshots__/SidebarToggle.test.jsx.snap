// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly with close icon 1`] = `
<button
  aria-controls="sidebar"
  className="btn sidebar-toggle-btn px-1 mr-2 bg-white"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  <span
    aria-hidden={true}
    className="fa fa-times"
    id="Icon2"
  />
  <span
    className="sr-only"
  >
    close menu
  </span>
</button>
`;

exports[`<Sidebar /> renders correctly with menu icon 1`] = `
<button
  aria-controls="sidebar"
  className="btn sidebar-toggle-btn px-1 mr-2 bg-white"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  <span
    aria-hidden={true}
    className="fa fa-bars"
    id="Icon1"
  />
  <span
    className="sr-only"
  >
    open menu
  </span>
</button>
`;
